### PR TITLE
HPCC-16981 Ensure StringAttr::set is safe when called with *this

### PR DESCRIPTION
**Pull request by ghalliday**
Signed-off-by: Gavin Halliday <gavin.halliday@lexisnexis.com>

### CODE REVIEW

**Comment by ghalliday** [comment]
@RussWhitehead please check the ldap change
@richardkchapman please review.

File: system/jlib/jstring.cpp [hunk 1]
```diff
@@ -1318,24 +1318,25 @@ StringAttr& StringAttr::operator = (StringAttr && from)
...
 {
-  if (text)
-    free(text);
-  text = (char *)malloc(_len+1);
-  memcpy(text, _text, _len);
-  text[_len] = 0;
+    char * oldtext = text;
+    text = (char *)malloc(_len+1);
+    memcpy(text, _text, _len);
+    text[_len] = 0;
```
Comment (RussWhitehead) [inline system/jlib/jstring.cpp]: curious why previous 3 lines are better than just calling strdup

**Review by RussWhitehead — APPROVED** [review]
One minor question, otherwise looks good

File: system/jlib/jstring.cpp (hunk 1 above)
Comment (richardkchapman) [inline system/jlib/jstring.cpp]: Because it's not null-terminated

**Pull request merged**